test/test_nn.py

```diff
@@ -2505,6 +2526,131 @@ def test_embedding_bag(self):
         self._test_EmbeddingBag(False, 'sum', True)
         self._test_EmbeddingBag(False, 'mean', True)
 
+    @staticmethod
+    def _get_reduction(mode='mean', dim=0):
+        def safe_mean(tensor, dim):
+            if tensor.size(dim) > 0:
+                return tensor.mean(dim)
+            return torch.zeros_like(tensor.mean(dim))
+
+        def safe_max(tensor, dim):
+            if tensor.size(dim) > 0:
+                return tensor.max(dim)[0]
+            return torch.zeros_like(tensor.mean(dim))
+
+        if mode == 'mean':
+            return lambda t: safe_mean(t, dim)
+        elif mode == 'sum':
+            return lambda t: t.sum(dim)
+        elif mode == 'max':
+            return lambda t: safe_max(t, dim)
+        else:
+            assert false
+
+    @staticmethod
+    def _embedding_bag_reference(input, weight, offsets=None, mode='mean',
```

Comment:
I don't understand the purpose of this function

Reply:
It's a reference python implementation of embedding_bag

Reply:
Ah, ok, I thought it had something to do with C++ references. I'd call "reference_implementation" just to be sure and add some comments.